test/Core.Test/Services/SendServiceTests.cs

```diff
@@ -0,0 +1,377 @@
+        {
+            var initialSendDict = initialSends.ToDictionary(s => s.Id, s => s);
+            sutProvider.GetDependency<IUserService>().GetUserIdAsync().Returns(userId);
+            sutProvider.GetDependency<IStorageService>().GetAsync<Dictionary<string, SendData>>(SEND_KEY(userId)).Returns(initialSendDict);
+
+            var updatedSends = CoreHelpers.Clone(initialSendDict);
+            foreach (var kvp in updatedSends)
+            {
+                kvp.Value.Disabled = !kvp.Value.Disabled;
+            }
+
+            await sutProvider.Sut.UpsertAsync(updatedSends.Values.ToArray());
+
+            Predicate<Dictionary<string, SendData>> matchSendsPredicate = actual =>
+            {
+                Assert.Equal(updatedSends.Count, actual.Count);
+                foreach (var (expectedKvp, actualKvp) in updatedSends.Zip(actual))
+                {
+                    Assert.Equal(expectedKvp.Key, actualKvp.Key);
+                    TestHelper.AssertPropertyEqual(expectedKvp.Value, actualKvp.Value);
+                }
+                return true;
+            };
+            await sutProvider.GetDependency<IStorageService>().Received(1).SaveAsync(SEND_KEY(userId), Arg.Is<Dictionary<string, SendData>>(d => matchSendsPredicate(d)));
+        }
+
+        [Theory]
+        [InlineCustomAutoData(new[] { typeof(SutProviderCustomization), typeof(TextSendCustomization) })]
+        [InlineCustomAutoData(new[] { typeof(SutProviderCustomization), typeof(FileSendCustomization) })]
+        public async Task UpsertAsync_NewSends_Success(SutProvider<SendService> sutProvider, string userId, IEnumerable<SendData> initialSends, IEnumerable<SendData> newSends)
+        {
+            var initialSendDict = initialSends.ToDictionary(s => s.Id, s => s);
+            sutProvider.GetDependency<IUserService>().GetUserIdAsync().Returns(userId);
+            sutProvider.GetDependency<IStorageService>().GetAsync<Dictionary<string, SendData>>(SEND_KEY(userId)).Returns(initialSendDict);
+
+            var expectedDict = CoreHelpers.Clone(initialSendDict).Concat(newSends.Select(s => new KeyValuePair<string, SendData>(s.Id, s)));
+
+            await sutProvider.Sut.UpsertAsync(newSends.ToArray());
+
+            Predicate<Dictionary<string, SendData>> matchSendsPredicate = actual =>
+            {
+                Assert.Equal(expectedDict.Count(), actual.Count);
+                foreach (var (expectedKvp, actualKvp) in expectedDict.Zip(actual))
+                {
+                    Assert.Equal(expectedKvp.Key, actualKvp.Key);
+                    TestHelper.AssertPropertyEqual(expectedKvp.Value, actualKvp.Value);
+                }
+                return true;
+            };
+            await sutProvider.GetDependency<IStorageService>().Received(1).SaveAsync(SEND_KEY(userId), Arg.Is<Dictionary<string, SendData>>(d => matchSendsPredicate(d)));
+        }
+
+        // TODO test encrypt
```

Comment:
It looks like you are testing encrypt, if I'm not mistaken, you can probably remove the `// TODO` here.